### PR TITLE
Feat: add JFrog webhook trigger

### PR DESCRIPTION
**Pull request by chwetion**
### Description of your changes
add JFrog webhook trigger: #3029

### Special notes for your reviewer
jfrog webhook payload have no create time, image url and private/public fileds, use imageURL field in request header to mark image url, if not set request header in jfrog webhook configuration, just use `reponame/imagename:tag` to update application image field

### CODE REVIEW

**Review by wonderflow — APPROVED** [review]
LGTM

**Comment by wonderflow** [comment]
please also add docs in kubevela.io, thanks

File: pkg/apiserver/rest/apis/v1/types.go [hunk 1]
```diff
@@ -464,6 +464,24 @@ type DockerHubRepository struct {
...
+// HandleApplicationTriggerJFrogRequest application trigger JFrog webhook request
+type HandleApplicationTriggerJFrogRequest struct {
+	Domain    string           `json:"domain"`
+	EventType string           `json:"event_type"`
+	Data      JFrogWebhookData `json:"data"`
+}
+
+// JFrogWebhookData is the data of JFrog webhook request
+type JFrogWebhookData struct {
+	URL       string
```
Comment (barnettZQG) [inline pkg/apiserver/rest/apis/v1/types.go:476]: JSON tag?

Comment (chwetion) [reply pkg/apiserver/rest/apis/v1/types.go:476]: it's not parse from request body, i am not sure need json tag.

File: pkg/apiserver/rest/usecase/webhook.go [hunk 2]
```diff
@@ -423,3 +429,74 @@ func (c *harborHandlerImpl) handle(ctx context.Context, webhookTrigger *model.Ap
...
+	if err != nil {
+		return nil, err
+	}
+	if len(comps) == 0 {
+		return nil, bcode.ErrApplicationComponetNotExist
+	}
+
+	// use the first component as the target component
+	component := comps[0].(*model.ApplicationComponent)
+	image := fmt.Sprintf("%s/%s:%s", jfrogReq.Data.RepoKey, jfrogReq.Data.ImageName, jfrogReq.Data.Tag)
```
Comment (barnettZQG) [inline pkg/apiserver/rest/usecase/webhook.go:469]: Need a RepoKey?

File: pkg/apiserver/rest/usecase/webhook.go [hunk 3]
```diff
@@ -423,3 +429,74 @@ func (c *harborHandlerImpl) handle(ctx context.Context, webhookTrigger *model.Ap
...
+	if jfrogReq.Data.URL != "" {
```
Comment (barnettZQG) [inline pkg/apiserver/rest/usecase/webhook.go:470]: https://www.jfrog.com/confluence/display/JFROG/Webhooks#Webhooks-Domain:Docker 

The URL is not param?

File: pkg/apiserver/rest/usecase/webhook.go [hunk 4]
```diff
@@ -423,3 +429,74 @@ func (c *harborHandlerImpl) handle(ctx context.Context, webhookTrigger *model.Ap
...
+func (c *webhookUsecaseImpl) newJFrogHandler(req *restful.Request) (webhookHandler, error) {
+	var jfrogReq apisv1.HandleApplicationTriggerJFrogRequest
+	if err := req.ReadEntity(&jfrogReq); err != nil {
+		return nil, bcode.ErrInvalidWebhookPayloadBody
+	}
+	if jfrogReq.Domain != model.JFrogDomainDocker || jfrogReq.EventType != model.JFrogEventTypePush {
+		return nil, bcode.ErrInvalidWebhookPayloadBody
+	}
+	// jfrog should use request header to give URL, it is not exist in request body
+	jfrogReq.Data.URL = req.HeaderParameter("imageURL")
```
Comment (barnettZQG) [inline pkg/apiserver/rest/usecase/webhook.go:447]: It's configured in the custom header?

Comment (chwetion) [reply pkg/apiserver/rest/usecase/webhook.go:447]: yes, configurated in jfrog webhook web

File: pkg/apiserver/rest/usecase/webhook.go (hunk 2 above)
Comment (chwetion) [reply pkg/apiserver/rest/usecase/webhook.go:469]: RepoKey/ImageName equivalent to FullName

File: pkg/apiserver/rest/usecase/webhook.go (hunk 4 above)
Comment (barnettZQG) [reply pkg/apiserver/rest/usecase/webhook.go:447]: Is it better X-JfrogURL?

**Review by barnettZQG — APPROVED** [review]
LGTM

**Review by FogDong — APPROVED** [review]
Cool!
Can you also add some `how to` docs in [kubevela.io](https://github.com/oam-dev/kubevela.io)?

**Comment by barnettZQG** [comment]
@chwetion Please rebase the branch.

**Pull request merged**